src/xbot/common/subsystems/ConnectionMonitorSubsystem.java

```diff
@@ -0,0 +1,42 @@
+package xbot.common.subsystems;
+
+import com.google.inject.Singleton;
+import edu.wpi.first.wpilibj.Timer;
+import xbot.common.command.BaseSubsystem;
+import xbot.common.logic.Latch;
+import xbot.common.properties.DoubleProperty;
+import xbot.common.properties.XPropertyManager;
+
+import java.util.Observable;
+import java.util.Observer;
+
+@Singleton
+public class ConnectionMonitorSubsystem extends BaseSubsystem implements Observer {
+
+    protected final DoubleProperty timeOut;
+    private final Latch connectionLatch = new Latch(true, Latch.EdgeType.FallingEdge);
+
+    private double lastPacketReceivedTimestamp = Timer.getFPGATimestamp();
+    private double previousDisconnectionTimestamp = -1;
+
+    public ConnectionMonitorSubsystem(XPropertyManager propertyManager) {
+        log.info("Creating");
+        timeOut = propertyManager.createPersistentProperty("ConnectionMonitorTimeOut Seconds", 1.0);
+        connectionLatch.addObserver(this);
+    }
+
+    public synchronized void setLastPacketReceivedTimestamp(double currentTimestamp) {
```

Comment:
Why `synchronized`? I don't think there are other threads involved here...

Reply:
Original idea was anyone can call it but since only BaseRobot is calling it yeah it doesn't make sense for synchronized.